lib/commands/edit.js

```diff
@@ -2,7 +2,7 @@
 // open the package folder in the $EDITOR
 
 const { resolve } = require('path')
-const fs = require('graceful-fs')
```

Comment:
This is doing some important things, specifically:

> Queues up open and readdir calls, and retries them once something closes if there is an EMFILE error from too many file descriptors.
> retries reading a file if read results in EAGAIN error.

I don't know how it affects this command but we do need to be careful removing it.

Reply:
Yeah this is interesting because this is the only place we call graceful-fs directly. We also patch the entire fs object globally with graceful-fs but that only applies to callback style fs functions. So if I'm reading it right, for as long as we've been using fs.promises or fs/promises, we haven't been getting the benefits of graceful-fs.

Reply:
As for this command, it only runs a single lstat call in a command without much other file IO so I think it's ok.

I should run some tests and just see how often graceful-fs patched functions are called in a typical install or other IO heavy command.

Reply:
> but that only applies to callback style fs functions

Ah.

> I should run some tests and just see how often graceful-fs patched functions are called in a typical install or other IO heavy command.

Yeah I think it was reify that was the big one.